### DISCUSSION TITLE
Option to make links and paths relative

### DISCUSSION THREAD

**Discussion by JoshuaDoust** [retypeapp/retype#133] [Ideas]
Hi, I'm working with a website which, after being generated by Retype, would be deployed to one of a number of destinations on the server. This would be done in an automated manner, so changing the retype.yaml file for each target is not practical. Unfortunately, it appears that all links within the generated files are absolute paths rather than relative ones, so having a configurable deployment in this manner doesn't work properly.

My feature request is to have an option for Retype to generate all paths as relative so that the site can be deployed to any location without modification.

Also, if there's any way of accomplishing my desired result within existing Retype functionality, I'd appreciate any guidance.

Thanks.

**Comment by geoffreymcgill** [comment]
Just to confirm, are you rebuilding the project for each separate deployment but do not want to change the `retype.yml` config for each deployment? Or, are you just building once and then deploying that built website to multiple locations, potentially under varying root paths?

**Reply by JoshuaDoust** [reply]
Ideally I'd like to do the latter, to build once then deploy to a number of locations with different root paths.

**Reply by geoffreymcgill** [reply]
We discussed this today and we think it _might_ be possible. Interestingly, Retype was first built with the requirement that all links were generated with relative paths, exactly as per your requirement to enable deploying the built site to any folder or sub-folder. 

At some point we ran into a problem with supporting only relative pathing and had to revert to having the `base` config. As of `v1.8.0`, the `base` config was simplified into the [`url`](https://retype.com/configuration/project/#url) config and a separate `base` is no longer required.

We have to dig deeper and will run some experiments. I will keep this thread updated with our progress, although this is a relatively large change and is going to take some time to get working, but we do want to see this working.

Thanks for the feature request.

**Reply by ApolonTorq** [reply]
> At some point we ran into a problem with supporting only relative pathing and had to revert to having the base config. As of v1.8.0, the base config was simplified into the url config and a separate base is no longer required.

Relative paths are a pretty fundamental basis for all forms of content systems and hierarchical systems in general. I'd urge you to revisit the decision to generate absolute paths across the board.

Is the `retype build --override` option applicable in this scenario?
I haven't experimented with it and haven't found any docs on the syntax for that command line option. Is it a whole file override or can you override individual config values?

**Reply by geoffreymcgill** [reply]
> Is the `retype build --override` option applicable in this scenario?

Yes, [`--override`](http://localhost:5001/guides/cli/#usage-of---override) might be able to help. Individual config values can be overridden, including [`url`](http://localhost:5001/configuration/project/#url).

We have yet to get to the feature of outputting relative paths. We will continue to keep this thread updated with our progress.
